ur_robot_driver/src/dashboard_client_ros.cpp

```diff
@@ -312,20 +296,17 @@ bool DashboardClientROS::handleRunningQuery(const ur_dashboard_msgs::srv::IsProg
 bool DashboardClientROS::handleSavedQuery(ur_dashboard_msgs::srv::IsProgramSaved::Request::SharedPtr req,
                                           ur_dashboard_msgs::srv::IsProgramSaved::Response::SharedPtr resp)
 {
-  try {
-    resp->answer = this->client_.sendAndReceive("isProgramSaved\n");
-    std::regex expected("(true|false) ([^\\s]+)");
-    std::smatch match;
-    resp->success = std::regex_match(resp->answer, match, expected);
-
-    if (resp->success) {
-      resp->program_saved = (match[1] == "true");
-      resp->program_name = match[2];
-    }
-  } catch (const urcl::UrException& e) {
-    RCLCPP_ERROR(rclcpp::get_logger("Dashboard_Client"), "Service Call failed: '%s'", e.what());
-    resp->answer = e.what();
-    resp->success = false;
+  auto dashboard_response =
+      dashboardCallWithChecks([this]() { return client_->commandIsProgramSavedWithResponse(); }, resp);
+  if (resp->success) {
+    handleDashboardResponseData(
+        [dashboard_response, resp]() {
```

Comment:
I think it would increase the readability if this lambda is defined within the function, but outside the function call.